From 5f00c70086fd36bcb524b0ef97736ec23e426e82 Mon Sep 17 00:00:00 2001
From: Matej Habrnal <mhabrnal@redhat.com>
Date: Thu, 10 Nov 2016 17:06:04 +0100
Subject: [PATCH] cli: do not ask for password if PrivateReports = no

abrt-cli list said "PrivateReports is enabled ..." and ask for root password
even if PrivateReports is disabled.

Related to #1318354

Signed-off-by: Matej Habrnal <mhabrnal@redhat.com>
Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/cli/abrt-cli-core.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/cli/abrt-cli-core.c b/src/cli/abrt-cli-core.c
index 536ef63..74fc43a 100644
--- a/src/cli/abrt-cli-core.c
+++ b/src/cli/abrt-cli-core.c
@@ -85,6 +85,12 @@ void restart_as_root_if_needed(unsigned cmd_argc, const char *cmd_argv[])
     if (g_settings_privatereports && getuid() == 0)
         return;
 
+    if (g_settings_privatereports == false)
+    {
+        log(_("PrivateReports is disabled. Run abrt-cli-root to see all problems detected by ABRT."));
+        return;
+    }
+
     log(_("PrivateReports is enabled. Only \"root\" can see the problems detected by ABRT."));
 
     if (!ask_yes_no(_("Do you wan to run abrt-cli-root?")))
-- 
1.7.1

